akka-http-scala/src/main/scala/akka/http/scaladsl/server/directives/FileAndResourceDirectives.scala

```diff
@@ -208,6 +202,35 @@ object FileAndResourceDirectives extends FileAndResourceDirectives {
     rec(if (path.startsWithSlash) path.tail else path)
   }
 
+  object ResourceFile {
+    def apply(url: URL): Option[ResourceFile] = url.getProtocol match {
+      case "file" ⇒
+        val file = new File(url.toURI)
+        if (file.isDirectory) None
+        else Some(ResourceFile(url, file.length(), file.lastModified()))
+      case "jar" ⇒
+        val jarFile = url.getFile
+        val startIndex = if (jarFile.startsWith("file:")) 5 else 0
+        val bangIndex = jarFile.indexOf("!")
+        val jarFilePath = jarFile.substring(startIndex, bangIndex)
+        val resourcePath = jarFile.substring(bangIndex + 2)
+        val jar = new java.util.zip.ZipFile(jarFilePath)
+        try {
+          val entry = jar.getEntry(resourcePath)
+          if (entry.isDirectory) None
+          else {
+            // isDirectory returns false for URLs pointing to directories without trailing slash
```

Comment:
Yes, no surprise because it's implemented as `name.endsWith("/");`. I'm not sure if there have to be entries included for directories in the zip file format at all or if they are synthetic.